### PR TITLE
Implement `performActions` for devtools automation protocol

### PR DESCRIPTION
**Pull request by christian-bromann**
## Proposed changes

These are the last two important protocol commands we have to implement for the devtools automation protocol. Given that there are a few commands (e.g. `keys` or `dragAndDrop`) that use `performActions` this was necessary to do before the v6 release.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have added proper type definitions for new commands (if appropriate)

### CODE REVIEW

File: e2e/element.test.js [hunk 1]
```diff
@@ -120,6 +120,66 @@ describe('elements', () => {
...
+                button: 0
+            }]
+        }])
+
+        const elem = await browser.findElement('css selector', '.searchinput')
+        expect(await browser.getElementProperty(elem[ELEMENT_KEY], 'value'))
+            .toBe('Dropped!')
+    })
+
+    it('should be able to use keys command', async () => {
```
Comment (erwinheitzman) [inline e2e/element.test.js]: Mind if we also test keys like SHIFT and an emoji?

File: packages/devtools/src/commands/performActions.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                        x += lastPointer.x
+                        y += lastPointer.y
+                    }
+
+                    lastPointer.x = x
+                    lastPointer.y = y
+                    await keyboardFn(x, y, { steps: 10 })
+                    continue
+                } else {
+                    const pptrButton = (
```
Comment (erwinheitzman) [inline packages/devtools/src/commands/performActions.js]: I would make 'left' the default.
I would also prefer a `switch` or some `if` statements for readability.

File: packages/devtools/src/devtoolsdriver.js [hunk 3]
```diff
@@ -103,9 +103,11 @@ export default class DevToolsDriver {
             }
 
             this.emit('result', { command, params, retries, result })
-            log.info('RESULT', command.toLowerCase().includes('screenshot')
-                && typeof result === 'string' && result.length > 64
-                ? `${result.substr(0, 61)}...` : result)
+            if (typeof result !== 'undefined') {
```
Comment (erwinheitzman) [inline packages/devtools/src/devtoolsdriver.js]: I would like the readability improved here as well by splitting the condition or use variables/functions for naming perhaps

File: packages/devtools/src/commands/performActions.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    const lastPointer = {}
+
+    /**
+     * see https://github.com/jlipps/simple-wd-spec#input-sources-and-corresponding-actions
+     * for details on the `actions` format
+     */
+    for (const action of actions) {
+        if (action.type === null || action.type === 'null') {
+            for (const singleAction of action.actions) {
+                await sleep(singleAction.duration)
```
Comment (erwinheitzman) [inline packages/devtools/src/commands/performActions.js]: `singleAction.duration` can be undefined here?

File: e2e/element.test.js (hunk 1 above)
Comment (christian-bromann) [reply e2e/element.test.js]: I added a test that includes an emoji. It seems that using shift to get characters written as uppercase seems to be not possible. I would maybe create a bug to investigate this later but wouldn't see it as a show stopper.

File: packages/devtools/src/commands/performActions.js (hunk 2 above)
Comment (christian-bromann) [reply packages/devtools/src/commands/performActions.js]: I added a comment for better readability and made `left` the default value. Using switch here doesn't seem to make the code more readable IMO.

File: packages/devtools/src/devtoolsdriver.js (hunk 3 above)
Comment (christian-bromann) [reply packages/devtools/src/devtoolsdriver.js]: created an `isScreenshot` variable to make it more readable.

File: packages/devtools/src/commands/performActions.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+        if (action.type === 'key') {
+            const skipChars = []
+            for (const singleAction of action.actions) {
+                if (singleAction.type === 'pause') {
+                    await sleep(singleAction.duration)
+                    continue
+                }
+
+                const cmd = singleAction.type.slice(3).toLowerCase()
```
Comment (mgrybyk) [inline packages/devtools/src/commands/performActions.js]: What if we avoid using constants like 3 or 7 and use action.type length instead or something like that to make it more understandable? Maybe define function to get cmd based on action.type

Same for pointer

File: packages/devtools/src/commands/performActions.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+                const keyboardFn = ::page.keyboard[cmd]
```
Comment (mgrybyk) [inline packages/devtools/src/commands/performActions.js]: should we check if cmd is one of type/press/down/up/sendCharacter ?

ref https://github.com/puppeteer/puppeteer/blob/v2.1.1/docs/api.md#class-keyboard

**Comment by erwinheitzman** [comment]
I think Mykola has some valid points. Should we address these?

**Comment by christian-bromann** [comment]
> Should we address these?

Absolutely, just didn't have the time to circle back to this.

File: packages/devtools/src/commands/performActions.js (hunk 6 above)
Comment (christian-bromann) [reply packages/devtools/src/commands/performActions.js]: I use either `down`/`up` or `sendCharacter` as I thought they would best represent what the protocol is requesting.

**Comment by christian-bromann** [comment]
@erwinheitzman @mgrybyk addressed both comments, anything left to do here?

**Pull request merged**